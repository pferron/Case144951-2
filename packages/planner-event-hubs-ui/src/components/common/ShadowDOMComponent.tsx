import React, { useEffect, useRef } from 'react';
import { injectTestId } from '@cvent/nucleus-test-automation';

interface ShadowDOMComponentProps {
  id: string;
  customHtml: string;
  customCss: string;
  customJs: string;
  shadowMode?: 'open' | 'closed';
}

const scriptToStopOpeningLinksInIframe = `
  document.querySelector('#custom-header').shadowRoot.addEventListener('click', event => {
    event.preventDefault();
  });
`;

function ShadowDOMComponent({ id, customHtml, customCss, customJs, shadowMode = 'open' }: ShadowDOMComponentProps) {
  const shadowRootRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const shadowRoot = shadowRootRef.current?.attachShadow({ mode: shadowMode }) ?? null;
    if (shadowRoot) {
      const style = document.createElement('style');
      style.textContent = customCss;
      shadowRoot.appendChild(style);

      const parser = new DOMParser();
      const parsedHTML = parser.parseFromString(customHtml, 'text/html');
      shadowRoot.appendChild(parsedHTML.body);

      const script = document.createElement('script');
      script.textContent = customJs;
      shadowRoot.appendChild(script);

      const script2 = document.createElement('script');
      script2.textContent = scriptToStopOpeningLinksInIframe;
      shadowRoot.appendChild(script2);
    }
  }, [customHtml, customCss, shadowMode, customJs]);

  return <div {...injectTestId(id)} id={id} ref={shadowRootRef} />;
}

export default ShadowDOMComponent;
