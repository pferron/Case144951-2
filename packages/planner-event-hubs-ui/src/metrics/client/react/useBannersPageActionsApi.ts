/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface AssignPagesButtonClicked extends TrackProperties {}
export interface BannerContentSectionChecked extends TrackProperties {
  // Title provided for the banner
  bannerTitle: string;
  // Description provided for the banner
  bannerBody?: string;
  // Button text if there are any buttons added to the banner
  buttonText?: string;
  // Destination page or link on click on the button
  whereWillThisButtonBringYourMembers?: string;
  // Name of the page that button takes when selected Events+ hub
  selectAPage?: string;
  // External Link to which the button clicks takes to
  externalLink?: string;
}
export interface BannerDesignSectionChecked extends TrackProperties {
  // Text alignment selected for the banner
  textAlignment: string;
  // Font color of the text that appears on the banner
  fontColor?: string;
}
export interface BannerImageChecked extends TrackProperties {}
export interface BannerNameProvided extends TrackProperties {
  // Template chosen for the banner
  bannerTemplate: string;
  // Name provided for the banner
  bannerName: string;
}
export interface CreateBannerButtonClicked extends TrackProperties {}
export interface PagesAssignButtonClicked extends TrackProperties {
  // Name of the page assigned to the banner
  bannerPlacement: string;
}

interface AnalyticsApi {
  assignPagesButtonClicked: (
    props?: AssignPagesButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerContentSectionChecked: (
    props: BannerContentSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerDesignSectionChecked: (
    props: BannerDesignSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerImageChecked: (props?: BannerImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
  bannerNameProvided: (props: BannerNameProvided, options?: Rudder.Options, callback?: ApiCallback) => void;
  createBannerButtonClicked: (
    props?: CreateBannerButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  pagesAssignButtonClicked: (props: PagesAssignButtonClicked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBannersPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBannersPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who has intent to assign pages to banners in order to show banners on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : assign pages button clicked
   * downstreamName: assign pages button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign pages' in page section of banner
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {AssignPagesButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const assignPagesButtonClicked = useCallback(
    (props?: AssignPagesButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'assign pages button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'assign pages button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provides banner details in order to update the banner content section
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner content section checked
   * downstreamName: banner content section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner content section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerContentSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerContentSectionChecked = useCallback(
    (props: BannerContentSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner content section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner content section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who updates design section in order to change the text alignments in banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner design section checked
   * downstreamName: banner design section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner design section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerDesignSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerDesignSectionChecked = useCallback(
    (props: BannerDesignSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner design section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner design section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use in banners
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner image checked
   * downstreamName: banner image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in banners section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerImageChecked = useCallback(
    (props?: BannerImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provide a name to banners in order to create the banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner name provided
   * downstreamName: banner name provided
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerNameProvided} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerNameProvided = useCallback(
    (props: BannerNameProvided, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner name provided',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner name provided'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who wants to create banner in order to use in eventsplus pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : create banner button clicked
   * downstreamName: create banner button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when the user clicks on Create Banner' button on the banners page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {CreateBannerButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const createBannerButtonClicked = useCallback(
    (props?: CreateBannerButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'create banner button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'create banner button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who assigns the pages to banners in order to show the banner on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : pages assign button clicked
   * downstreamName: pages assign button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign' in the select 'Pages' modal
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {PagesAssignButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const pagesAssignButtonClicked = useCallback(
    (props: PagesAssignButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'pages assign button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'pages assign button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    assignPagesButtonClicked,
    bannerContentSectionChecked,
    bannerDesignSectionChecked,
    bannerImageChecked,
    bannerNameProvided,
    createBannerButtonClicked,
    pagesAssignButtonClicked
  };
};
