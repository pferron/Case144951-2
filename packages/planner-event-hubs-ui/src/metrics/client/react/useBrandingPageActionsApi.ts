/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface BrandingSaveButtonClicked extends TrackProperties {
  // Primary color of the events+ hub
  primaryColor: string;
  // Secondary color color of the events+ hub
  secondaryColor: string;
  // BackgroundColor color color of the events+ hub
  backgroundColor: string;
  // Mood of the events+ hub
  selectYourMood: string;
  // Whether safemode is turned on or off for the events+ hub
  safeColorMode: boolean;
  // Whether background color is overriden for the events+ hub
  overrideBackgroundColor?: boolean;
}
export interface LogoImageChecked extends TrackProperties {}

interface AnalyticsApi {
  brandingSaveButtonClicked: (
    props: BrandingSaveButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  logoImageChecked: (props?: LogoImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBrandingPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBrandingPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who makes changes in themes in order to change the colors of branding
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : branding save button clicked
   * downstreamName: branding save button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when clicks on save button in brandind edit page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BrandingSaveButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const brandingSaveButtonClicked = useCallback(
    (props: BrandingSaveButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'branding save button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'branding save button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use as logo in events+
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : logo image checked
   * downstreamName: logo image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in branding section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {LogoImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const logoImageChecked = useCallback(
    (props?: LogoImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'logo image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'logo image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    brandingSaveButtonClicked,
    logoImageChecked
  };
};
