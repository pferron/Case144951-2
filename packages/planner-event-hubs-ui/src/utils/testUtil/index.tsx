import { render, configure } from '@testing-library/react';
import { TestWrapper } from './TestWrapper';

configure({ testIdAttribute: 'data-cvent-id' });

type RenderParams = Parameters<typeof render>;
type RenderReturn = ReturnType<typeof render>;
const customRender = (ui: RenderParams[0], options?: RenderParams[1]): RenderReturn =>
  render(ui, { wrapper: TestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
