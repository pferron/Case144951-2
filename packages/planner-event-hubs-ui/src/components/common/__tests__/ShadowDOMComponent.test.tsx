import { render, screen } from '@utils/testUtil';
import React from 'react';
import { waitFor } from '@testing-library/react';
import ShadowDOMComponent from '@components/common/ShadowDOMComponent';

const customHtml = `<div class="topnav" id="myTopnav">
  <a href="#home" class="active">Home</a>
  <a href="#news">News</a>
  <a href="#contact">Contact</a>
  <div class="dropdown">
    <button class="dropbtn">Dropdown 
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="#">Link 1</a>
      <a href="#">Link 2</a>
      <a href="#">Link 3</a>
    </div>
  </div> 
  <a href="#about">About</a>
 <a href="#side" onclick="openNav()">&#9776; side</span>
  <a href="javascript:void(0);" style="font-size:15px;" class="icon" onclick="myFunction()">&#9776;</a>
<div id="mySidenav" class="sidenav">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
  <a href="#">About</a>
  <a href="#">Services</a>
  <a href="#">Clients</a>
  <a href="#">Contact</a>
</div>
</div>`;

const customCss = `
body {margin:0;font-family:Arial}

.topnav {
  overflow: hidden;
  background-color: #333;
  z-index: 10001;
}

.topnav a {
  float: left;
  display: block;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;
}

.active {
  background-color: #04AA6D;
  color: white;
}

.topnav .icon {
  display: none;
}

.dropdown {
  float: left;
  overflow: hidden;
}

.dropdown .dropbtn {
  font-size: 17px;    
  border: none;
  outline: none;
  color: white;
  padding: 14px 16px;
  background-color: inherit;
  font-family: inherit;
  margin: 0;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 10001;
}

.dropdown-content a {
  float: none; font-family: 'KayPhoDu','Arial','Rubik','Helvetica','Arial';
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.topnav a:hover, .dropdown:hover .dropbtn {
  background-color: #555;
  color: white;
}

.dropdown-content a:hover {
  background-color: #ddd;
  color: black;
}

.dropdown:hover .dropdown-content {
  display: block;
}

@media screen and (max-width: 600px) {
  .topnav a:not(:first-child), .dropdown .dropbtn {
    display: none;
  }
  .topnav a.icon {
    float: right;
    display: block;
  }
}

@media screen and (max-width: 600px) {
  .topnav.responsive {position: relative;}
  .topnav.responsive .icon {
    position: absolute;
    right: 0;
    top: 0;
  }
  .topnav.responsive a {
    float: none;
    display: block;
    text-align: left;
  }
  .topnav.responsive .dropdown {float: none;}
  .topnav.responsive .dropdown-content {position: relative;}
  .topnav.responsive .dropdown .dropbtn {
    display: block;
    width: 100%;
    text-align: left;
  }
}  

.sidenav {
  height: 100%;
  width: 0;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  background-color: #111;
  overflow-x: hidden;
  transition: 0.5s;
  padding-top: 60px;
}

.sidenav a {
  padding: 8px 8px 8px 32px;
  text-decoration: none;
  font-size: 25px;
  color: #818181;
  display: block;
  transition: 0.3s;
}

.sidenav a:hover {
  color: #f1f1f1;
}

.sidenav .closebtn {
  position: absolute;
  top: 0;
  right: 25px;
  font-size: 36px;
  margin-left: 50px;
}

@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}
`;

const customJs = `
function myFunction() {
  var x = document.querySelector('#custom-header').shadowRoot.getElementById("myTopnav");
  if (x.className === "topnav") {
    x.className += " responsive";
  } else {
    x.className = "topnav";
  }
}   

function openNav() {
  document.querySelector('#custom-header').shadowRoot.getElementById("mySidenav").style.width = "250px";
}

function closeNav() {
 document.querySelector('#custom-header').shadowRoot.getElementById("mySidenav").style.width = "0";
}
`;

describe('Top Navigation', () => {
  it('renders custom component successfully', async () => {
    const { baseElement } = render(
      <ShadowDOMComponent id="custom-header" customHtml={customHtml} customCss={customCss} customJs={customJs} />
    );

    expect(baseElement).toMatchSnapshot();
    expect(await screen.findByTestId('custom-header')).toBeInTheDocument();

    await waitFor(
      async () => {
        // Had do this as shadow component are not accessible directly in regular dom
        // eslint-disable-next-line testing-library/no-node-access
        expect(document.querySelector('#custom-header').shadowRoot.getElementById('mySidenav')).toBeInTheDocument();
      },
      {
        timeout: 1000,
        interval: 100
      }
    );

    // This snapshot should not fail. Please do check the reason before updating it.
    // eslint-disable-next-line testing-library/no-node-access
    expect(document.querySelector('#custom-header').shadowRoot.getElementById('myTopnav')).toMatchSnapshot();
  });
});
