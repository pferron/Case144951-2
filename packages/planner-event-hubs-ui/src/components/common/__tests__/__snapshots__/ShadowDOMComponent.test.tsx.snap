// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Navigation renders custom component successfully 1`] = `
<body>
  <div>
    <div
      data-cvent-id="custom-header"
      id="custom-header"
    />
  </div>
</body>
`;

exports[`Top Navigation renders custom component successfully 2`] = `
<div
  class="topnav"
  id="myTopnav"
>
  
  
  <a
    class="active"
    href="#home"
  >
    Home
  </a>
  
  
  <a
    href="#news"
  >
    News
  </a>
  
  
  <a
    href="#contact"
  >
    Contact
  </a>
  
  
  <div
    class="dropdown"
  >
    
    
    <button
      class="dropbtn"
    >
      Dropdown 
      
      <i
        class="fa fa-caret-down"
      />
      
    
    </button>
    
    
    <div
      class="dropdown-content"
    >
      
      
      <a
        href="#"
      >
        Link 1
      </a>
      
      
      <a
        href="#"
      >
        Link 2
      </a>
      
      
      <a
        href="#"
      >
        Link 3
      </a>
      
    
    </div>
    
  
  </div>
   
  
  <a
    href="#about"
  >
    About
  </a>
  
 
  <a
    href="#side"
    onclick="openNav()"
  >
    ☰ side
  
  </a>
  <a
    class="icon"
    href="javascript:void(0);"
    onclick="myFunction()"
    style="font-size:15px;"
  >
    ☰
  </a>
  

  <div
    class="sidenav"
    id="mySidenav"
  >
    
  
    <a
      class="closebtn"
      href="javascript:void(0)"
      onclick="closeNav()"
    >
      ×
    </a>
    
  
    <a
      href="#"
    >
      About
    </a>
    
  
    <a
      href="#"
    >
      Services
    </a>
    
  
    <a
      href="#"
    >
      Clients
    </a>
    
  
    <a
      href="#"
    >
      Contact
    </a>
    

  </div>
  

</div>
`;
